src/main/java/com/example/ajaxonboardingproject/service/mapper/MovieSessionMapper.kt

```diff
@@ -7,24 +7,23 @@ import com.example.ajaxonboardingproject.service.CinemaHallService
 import com.example.ajaxonboardingproject.service.MovieService
 import org.springframework.stereotype.Component
 
+fun MovieSessionMapper.mapToModel(dto: MovieSessionRequestDto): MovieSession {
+    return MovieSession(
+            movie = this.movieService.get(dto.movieId),
+            cinemaHall = this.cinemaHallService.get(dto.cinemaHallId),
+            showTime = dto.showTime)
+}
+
+fun MovieSessionMapper.mapToDto(model : MovieSession): MovieSessionResponseDto {
+    return MovieSessionResponseDto(
+            movieSessionId = model.id!!,
+            cinemaHallId = model.cinemaHall.id!!,
+            movieTitle = model.movie.title,
+            movieId = model.movie.id!!,

```

Comment:
could we change `id` in all models to lateinit?

```kotlin
class Some {
   lateinit var lazyInitProp: String
}

Some().lazyInitProp // throws, because not initialized

val s = Some().apply {
   lazyInitProp = ""
}
s.lazyInitProp // ok
```

Reply:
yes it would be a great solution to !! but as far as i know we can not do primitive (Long) a lateinit

Reply:
What if we try to force kotlin using boxed long?

https://discuss.kotlinlang.org/t/force-hint-of-int-to-java-lang-integer/2441

Reply:
Thanks for help)